rules/consistent-destructuring.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const fix = (fixer, source, objectPattern, memberExpression) => {
+	const member = source.getText(memberExpression.property);
+	const {properties} = objectPattern;
+	const declarations = properties.map(property => source.getText(property));
+	const fixings = [fixer.replaceText(memberExpression, member)];
+
+	// Member is not already destructured
+	if (!declarations.includes(member)) {
+		const lastProperty = properties[properties.length - 1];
+
+		if (lastProperty) {
+			fixings.push(fixer.insertTextAfter(lastProperty, `, ${member}`));
+		} else {
+			fixings.push(fixer.replaceText(objectPattern, `{${member}}`));
+		}
+	}
+
+	return fixings;
+};
+
+const create = context => {
+	const source = context.getSourceCode();
+	const declarations = new Map();
+
+	return {
+		VariableDeclarator(node) {
+			const {id, init} = node;
+
+			if (id.type === 'ObjectPattern' && init.type !== 'Literal') {
+				declarations.set(source.getText(init), id);
+			}
+		},
+		MemberExpression(node) {
+			const {parent, object} = node;
+
+			// Ignore member function calls
+			if (parent.type === 'CallExpression' && parent.callee === node) {
+				return;
+			}
+
+			const objectPattern = declarations.get(source.getText(object));
+
+			if (objectPattern) {
+				const isNested = parent.type === 'MemberExpression';
+
+				context.report({
+					node,
+					message: 'Use destructured variables over properties.',
```

Comment:
Use [`messageId`](https://eslint.org/docs/developer-guide/working-with-rules#messageids) instead of a `message`. See [prefer-flat-map](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/6f10722c2397d9734375cc5165a3675e07752982/rules/prefer-flat-map.js#L62) for an example.